File: mobile/app/src/main/java/com/praeter/ui/signup/plan/PlanActivity.java
package com.praeter.ui.signup.plan;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.widget.Button;
import android.widget.RadioButton;

import androidx.annotation.Nullable;

import com.praeter.R;
import com.praeter.data.remote.dto.User;
import com.praeter.navigator.Navigator;
import com.praeter.ui.base.SimpleActivity;

import org.parceler.Parcels;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

@SuppressLint("NonConstantResourceId")
public class PlanActivity extends SimpleActivity {

    public static final String EXTRA_USER = "EXTRA_USER";

    @BindView(R.id.rb_free_plan)
    RadioButton rbFreePlan;

    @BindView(R.id.rb_premium_plan)
    RadioButton rbPremiumPlan;

    @BindView(R.id.btn_continue)
    Button btnContinue;

    @Inject
    Navigator navigator;

    private User user;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_plan);

        ButterKnife.bind(this);

        getSupportActionBar().setTitle(R.string.title_activity_plan);

        navigator = new Navigator(PlanActivity.this);

        Bundle extras = getIntent().getExtras();

        if (null == extras) {
            return;
        }

        user = Parcels.unwrap(extras.getParcelable(EXTRA_USER));
    }


    @OnClick(R.id.btn_continue)
    void onContinueButtonClicked() {

        if (rbFreePlan.isChecked()) {
            user.setPremium(false);
            navigator.callSuccessfulSignUpActivity(user);
        }

        if (rbPremiumPlan.isChecked()) {
            user.setPremium(true);
            navigator.callPremiumPlanActivity(user);
        }

        finish();
    }

}

File: mobile/app/src/main/java/com/praeter/ui/signup/plan/PlanActivity.kt
package com.praeter.ui.signup.plan

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.RadioButton
import com.praeter.data.remote.dto.user.User
import com.praeter.databinding.ActivityPlanBinding
import com.praeter.navigator.Navigator
import com.praeter.ui.base.BaseActivity
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class PlanActivity : BaseActivity() {

    private var _viewBinding: ActivityPlanBinding? = null

    // This property is only valid between onCreateView and
    // onDestroyView.
    private val binding get() = _viewBinding!!

    var rbFreePlan: RadioButton? = null
    var rbPremiumPlan: RadioButton? = null
    var btnContinue: Button? = null

    private var user: User? = null

   val navigator: Navigator = Navigator(this@PlanActivity)


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        _viewBinding = ActivityPlanBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val extras: Bundle = intent.extras ?: return
        user = extras.getParcelable(EXTRA_USER)
    }

    fun continueClick(view: View) {

        if (rbFreePlan!!.isChecked) {
            user?.isPremium = false
            navigator.callSuccessfulSignUpActivity(user)
        }
        if (rbPremiumPlan!!.isChecked) {
            user?.isPremium = true
            navigator.callPremiumPlanActivity(user)
        }
        finish()
    }

    companion object {
        const val EXTRA_USER = "EXTRA_USER"
    }
}